src/main/java/com/vaadin/demo/views/FixedNavStickyFooterView.java

```diff
@@ -0,0 +1,74 @@
+package com.vaadin.demo.views;
+
+import com.vaadin.demo.MainView;
+import com.vaadin.demo.components.ExAddContent;
+import com.vaadin.flow.component.Tag;
+import com.vaadin.flow.component.button.Button;
+import com.vaadin.flow.component.dependency.JsModule;
+import com.vaadin.flow.component.html.Anchor;
+import com.vaadin.flow.component.html.Div;
+import com.vaadin.flow.component.html.H1;
+import com.vaadin.flow.component.html.Header;
+import com.vaadin.flow.component.html.Paragraph;
+import com.vaadin.flow.component.polymertemplate.Id;
+import com.vaadin.flow.component.polymertemplate.PolymerTemplate;
+import com.vaadin.flow.component.textfield.TextField;
+import com.vaadin.flow.router.PageTitle;
+import com.vaadin.flow.router.Route;
+import com.vaadin.flow.router.RouterLink;
+import com.vaadin.flow.templatemodel.TemplateModel;
+
+@Route(value = "fixed-nav-sticky-footer", layout = MainView.class)
+@PageTitle("Fixed Nav & Sticky Footer")
+@Tag("fixed-nav-sticky-footer-view")
+@JsModule("./src/views/fixed-nav-sticky-footer-view.js")
+public class FixedNavStickyFooterView extends PolymerTemplate<TemplateModel> {
+
+    @Id("header")
+    Header header;
+
+    @Id("main")
+    Div main;
+
+    @Id("footer")
+    Div footer;
+
+    public FixedNavStickyFooterView() {
+        createHeader();
+
+        main.add(new H1("Sticky footer with fixed navbar"));
+        main.add(new Paragraph("Footer pinned to the bottom of the viewport."
+                + " If there is more content than fits the viewport, the footer"
+                + " will be pushed down and will be visible only when scrolled"
+                + " to the bottom of the page."));
+        main.add(new ExAddContent());
+
+        footer.add(new Paragraph("Place sticky footer content here."));
+    }
+
+    public void createHeader() {
+        RouterLink titleLink = new RouterLink("Fixed navbar", getClass());
+        titleLink.addClassName("titleLink");
+        header.add(titleLink);
+        header.add(new RouterLink("Home", getClass()));
+        header.add(new Anchor("#", "Link"));
+        Anchor a = new Anchor("#", "Help");
+        // Stretches this item in the flex container so that the following items
+        // will be at the right side of the screen.
+        a.getStyle().set("margin-right", "auto");
```

Comment:
If this is set as `margin-left` for `searchBlock` then the behaviour would be different when you make the viewport small enough so it wraps. Now it wraps so that search block goes to the left side when wrapping, but making it margin-left on it would make it always stay on the right side also after wrapping which doesn't look as nice imo.

Another option (to keep the current behaviour but having it separate from the links which might change) would be to add a dummy element in between like `<span style="flex: 1"></span>` (but creating it in Java).

I would probably prefer to keep it as it is since the comment explains it and it would be slightly more code to add an extra `Span` and I wouldn't want to complicate this part too much for so little benefit.

What do you think? I can add the Span if you think that's better.

Reply:
I see, of course it's better to only set it once. I'm ok to keep the styles on the anchor then.